dg/commands/cluster/install_assembly.py

```diff
@@ -35,53 +37,51 @@
         dg.config.cluster_credentials_manager.cluster_credentials_manager.get_current_credentials()
     )
 )
-@click.option("--server", required=True, help="OpenShift server URL")
-@click.option("--username", help="OpenShift username")
-@click.option("--password", help="OpenShift password")
-@click.option("--token", help="OpenShift OAuth access token")
-@click.option("--artifactory-user-name", required=True, help="Artifactory user name")
-@click.option("--artifactory-api-key", required=True, help="Artifactory API key")
-@click.option(
+@optgroup.group("Release build options")
```

Comment:
Everything looks great here, maybe one last optional proposal, but it's totally up to you if you want to do it, you have my approval either way:

My small proposal would be to pecify the option group for the shared options first and change the `use-dev` flag into something that fits the shared option group (e.g. `build-type` which could be either `dev` or `release` but would default to `release` when unset)

The reason I'm proposing this, is that exclusiveness between the two fields can then be defined as the help text of this parameter, so to given an example, in pseudo `--help` text:

```
Options:
  --server TEXT                   OpenShift server URL  [required]
  --username TEXT                 OpenShift username
  --password TEXT                 OpenShift password
  --token TEXT                    OpenShift OAuth access token
  --storage-class TEXT            Storage class used for installation
                                  [required]

  --version TEXT                  Cloud Pak for Data version
  --build-type TEXT               Build type to use. Can be "dev" or "release". Defaults to "release"
  --help                          Show this message and exit.
 Development build options
   --artifactory-user-name TEXT    Artifactory user name  [required]
   --artifactory-api-key TEXT      Artifactory API key  [required]
 Release build options
  --ibm-cloud-pak-for-data-entitlement-key TEXT
                                  IBM Cloud Pak for Data entitlement key
```

Again, this is just a cosmetic proposal, proceed as you wish :)